platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/SignedState.java

```diff
@@ -487,21 +488,22 @@ public long getMinRoundGeneration() {
     }
 
     /**
-     * Check if this is a state that needs to be eventually written to disk.
+     * Set the reason why this state needs to be written to disk.
      *
-     * @return true if this state eventually needs to be written to disk
+     * @param reason the reason why this state needs to be written to disk
      */
-    public boolean isStateToSave() {
-        return stateToSave;
+    public void setStateToDiskReason(@NonNull final StateToDiskReason reason) {
+        stateToDiskReason = reason;
     }
```

Comment:
Perhaps for a clearer API, we should so something like this. That way it is clear that a side effect of this method is that this state will be written to disk.

```suggestion
    public void markAsStateToSave(@NonNull final StateToDiskReason reason) {
        stateToDiskReason = reason;
    }
```

Reply:
done